core/templates/dev/head/galleries/Gallery.js

```diff
@@ -399,27 +401,39 @@ oppia.controller('Gallery', [
       }
     );
 
-    $scope.showFullGalleryGroup = function(galleryGroup) {
-      selectionDataService.clearCategories();
-      selectionDataService.addCategoriesToSelection(galleryGroup.categories);
-      // TODO(sll): is this line correct?
-      selectionDataService.clearLanguageCodes();
-      searchService.executeSearchQuery('', function() {
-        removeSplashCarousel();
-      });
-      // TODO(sll): Clear the search query from the search bar, too.
+    // Retrieves gallery data from the server.
+    $http.get(GALLERY_DATA_URL).success(function(data) {
+      $scope.currentUserIsModerator = Boolean(data.is_moderator);
+
+      // Note that this will cause an initial search query to be sent.
+      $rootScope.$broadcast(
+        'preferredLanguageCodesLoaded', data.preferred_language_codes);
+
+      if (data.username) {
+        if (urlService.getUrlParams().mode === 'create') {
+          $scope.showCreateExplorationModal(CATEGORY_LIST);
+        }
+      }
+    });
+
+    $scope.onRedirectToLogin = function(destinationUrl) {
+      siteAnalyticsService.registerStartLoginEvent('noSearchResults');
+      $timeout(function() {
+        $window.location = destinationUrl;
+      }, 150);
+      return false;
     };
   }
 ]);
 
 oppia.controller('SearchBar', [
-  '$scope', '$rootScope', 'searchService', 'oppiaDebouncer',
-  'ExplorationCreationButtonService', 'urlService', 'CATEGORY_LIST',
-  'selectionDataService',
+  '$scope', '$rootScope', '$timeout', '$window', 'searchService',
```

Comment:
Don't you still need selectionDataService here? It's used within the controller...